perllib/FixMyStreet/DB/Result/User.pm

```diff
@@ -391,4 +391,17 @@ sub update_reputation {
     $self->update;
 }
 
+sub categories {
+    my $self = shift;
+    return [] unless $self->get_extra_metadata('categories');
+    my @categories = $self->result_source->schema->resultset("Contact")->search({
+        id => $self->get_extra_metadata('categories'),
+    }, {
+        columns => [ 'category' ],
+        order_by => 'category',
+    })->all;
+    my $result = [ map( { $_->category } @categories ) ];
```

Comment:
You can probably use get_column here, something like `my @categories = search({ id => …}, { order_by => 'category' })->get_column('category')->all; my $result = \@categories;`